spark/dl/src/main/scala/com/intel/analytics/bigdl/parameters/AllReduceParameterManager.scala

```diff
@@ -0,0 +1,333 @@
+        innerLength))
+        }))
+    val gradientExecutorId = getGradientExecutorId()
+    val gradientExecutor = getLocalParameter[T](gradientExecutorId)
+    params.head.deCompress(gradientExecutor)
+  }
+
+  /**
+    * Use a fixed thread pool to launch a thread for each node of the weights. Each thread
+    * requests a node of the weights from the Spark block manager.
+    *
+    * @param localParameter The Tensor that will hold the retrieved weights.
+    * @return A [[FutureResult]] which contains a [[Future]] for each thread.
+    */
+  def getWeights[T: ClassTag](localParameter: Tensor[T]): FutureResult[Int] = {
+    val tasks = (0 until executorNum).map { pid =>
+      syncPool.submit {
+        new Callable[Int] {
+          override def call(): Int = {
+            try {
+              val blockId = getWeightBlockId(pid)
+              val localBuffer = BlockManagerWrapper.getLocalOrRemoteBytes(blockId).getOrElse {
+                throw new RuntimeException(s"Didn't find weight block $blockId in the block " +
+                  s"manager. Did you initialize this AllReduceParameter on every executor?")
+              }
+              val start = pid * taskSize + math.min(pid, extraSize)
+              val length = taskSize + (if (pid < extraSize) 1 else 0)
+              require(localBuffer.array().length == length * 2)
+              SerializerInstance.serialize(localBuffer).deCompress(0, localParameter, start, length)
+              BlockManagerWrapper.unlock(blockId)
+              pid
+            } catch {
+              case t: Throwable =>
+                logger.error("Error: " + ExceptionUtils.getStackTrace(t))
+                throw t
+            }
+          }
+        }
+      }
+    }
+    new FutureResult(tasks)
+  }
+
+  /**
+    * Put the portion of the weights that this node is responsible for to the block manager.
+    * Weights are placed locally, then pulled when needed by other nodes.
+    */
+  def sendWeightExecutor[T: ClassTag]() : Unit = {
+    val weightExecutorId = getWeightExecutorId()
+    val weightExecutor = getLocalParameter(weightExecutorId)
+    val blockId = getWeightBlockId(executorId)
+    BlockManagerWrapper.removeBlock(blockId)
+    val data = SerializerInstance.serialize(weightExecutor)
+    val block = BlockManagerWrapper.getLocalBytes(blockId)
+    if (block.isDefined) {
+      block.get.put(data.bytes())
+    } else {
+        BlockManagerWrapper.putBytes(blockId,
+          data.bytes(), StorageLevel.MEMORY_ONLY_SER)
+    }
+  }
+
+  /** Get a block from local blockmanager */
+  def getLocalParameter[T: ClassTag](blockId: BlockId): Tensor[T] = {
+    BlockManagerWrapper.getLocal(blockId)
+      .map(_.data.next().asInstanceOf[Tensor[T]])
+      .getOrElse(throw new IllegalStateException("Please initialize AllReduceParameter first!"))
+  }
+
+  /** Put a gradient in local blockmanager */
+  def sendGradients[T: ClassTag](gradient: Tensor[T], pid: Int): Unit = {
```

Comment:
Then the name and document really not reflect this difference. Can you fix this?